packages/cli/src/lib/spaces/outbound-rules.ts

```diff
@@ -1,6 +1,37 @@
 import {ux} from '@oclif/core'
 import * as Heroku from '@heroku-cli/schema'
 
+export function parsePorts(protocol: string, port: string | undefined) {
+  if (port === '-1' || port === 'any') {
+    if (protocol === 'icmp') {
+      return [0, 255]
+    }
+
+    return [0, 65535]
+  }
+
+  let actual: any[] = []
+  // eslint-disable-next-line no-eq-null, eqeqeq
+  if (port != null) {
+    const ports: any[] = port.split('-')
+    if (ports.length === 2) {
+      // eslint-disable-next-line unicorn/prefer-math-trunc
+      actual = [ports[0] | 0, ports[1] | 0]
+    } else if (ports.length === 1) {
+      // eslint-disable-next-line unicorn/prefer-math-trunc
+      actual = [ports[0] | 0, ports[0] | 0]
+    } else {
+      throw new Error('Specified --port range seems incorrect.')
+    }
+  }
```

Comment:
Are you sure this reproduces the original author's intent? I also thought this could be improved but with the bitwise operations on lines 19 and 22 I wasn't sure.

Reply:
Yes, the `bitwise or` operator will force a number cast and then truncate the float to an integer. It's a hack but typically not used since the nuances are not widely known and this use is outside anything covered in the docs. `Number.parseInt` does the same thing with the exception of `null` and `undefined` produce `NaN` versus `0` with the bitwise operators. Also, parseInt will correctly trim non alpha chars whereas bitwise does not. e.g.

```
'2abc' | 0 // outputs 0
parseInt('2abc') // outputs 2
```